runtime/src/main/kotlin/kotlin/native/concurrent/WorkerBoundReference.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+
+@SymbolName("Kotlin_WorkerBoundReference_create")
+external private fun createWorkerBoundReference(value: Any): NativePtr
+
+@SymbolName("Kotlin_WorkerBoundReference_deref")
+external private fun derefWorkerBoundReference(ref: NativePtr): Any
+
+/**
+ * A frozen shared reference to a Kotlin object
+ *
+ * Can be safely passed between workers, but can only be accessed on the worker it was created on,
+ * unless the referred object is frozen too
+ *
+ * Note: Garbage collector currently cannot free any reference cycles with [WorkerBoundReference] or [DisposableWorkerBoundReference] in them.
+ * Consider using [DisposableWorkerBoundReference] to manually resolve cycles with explicit calls to [DisposableWorkerBoundReference.dispose]
+ */
+@Frozen
+@NoReorderFields
+@ExportTypeInfo("theWorkerBoundReferenceTypeInfo")
+public class WorkerBoundReference<out T : Any>(value: T) {
+
+    private val ptr = createWorkerBoundReference(value)
+
+    /**
+     * The referenced value.
+     * @throws IncorrectDereferenceException if referred object is not frozen and this is called from a different worker, than [WorkerBoundReference] was created on
```

Comment:
`if referred object is not frozen and current worker is different from the one created [this]`
?